cost_test.go

```diff
@@ -118,121 +133,307 @@ func TestQueryComplexity(t *testing.T) {
 		Type: dataType,
 	})
 
-	deepDataInterface := NewInterface(InterfaceConfig{
-		Name:   "deepD",
-		Fields: deepDataFields,
-		ResolveType: func(p ResolveTypeParams) *Object {
-			return deepDataType
-		},
-	})
-
 	dataType.AddFieldConfig("iface", &Field{
 		Cost: 50,
 		Type: deepDataInterface,
 	})
 
+	queryCfg := ObjectConfig{
+		Name: "query",
+		Fields: Fields{
+			"example": &Field{
+				Type: dataType,
+			},
+		},
+	}
+
+	query := NewObject(queryCfg)
+	schema, err := NewSchema(SchemaConfig{
+		Query: query,
+	})
+	if err != nil {
+		t.Fatalf("Error in schema %v", err.Error())
+	}
+
 	tests := []struct {
 		description string
 		query       string
 		want        int
 	}{
+		// Note a test with an unused fragment isn't needed as that fails and a invalid query
 		{
 			description: "Simple Query",
-			query: `
-				  query Example($size: Int) {
-					a,
-					b
-				  }`,
+			query: `{
```

Comment:
Nit: These example queries have a lot of extra tabs pushing them way to the right.  You must have your tabsize set really small, like at 2.

Reply:
Here is what happened when debugging I didn't want to go past a ton of cases so I would comment these out in a block. That would push it all back and then when I uncommented it wouldn't quite return everything to the right place. I'll fix it, it is really ugly.